From e13f33483ec63f2319915214cc2805913def2663 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Thu, 28 Jun 2018 10:05:14 +0800
Subject: [PATCH 4144/5242] MLK-18609-6 usb: cdns3: gadget: need to prepare
 setup packet after status

commit  79589b75dbe1a0facb4eaaa8c283bde3bf27ffb1 from
https://source.codeaurora.org/external/imx/linux-imx.git

We need to prepare setup packet no matter configuration is set or
not.

Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/cdns3/gadget.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/cdns3/gadget.c b/drivers/usb/cdns3/gadget.c
index a4e390d..0d866dc 100644
--- a/drivers/usb/cdns3/gadget.c
+++ b/drivers/usb/cdns3/gadget.c
@@ -1392,7 +1392,6 @@ static int usb_ss_gadget_ep0_queue(struct usb_ep *ep,
 		if (!usb_ss->hw_configured_flag) {
 			gadget_writel(usb_ss, &usb_ss->regs->usb_conf,
 			USB_CONF__CFGSET__MASK); /* SET CONFIGURATION */
-			cdns_prepare_setup_packet(usb_ss);
 			gadget_writel(usb_ss, &usb_ss->regs->ep_cmd,
 				EP_CMD__ERDY__MASK | EP_CMD__REQ_CMPL__MASK);
 			/* wait until configuration set */
@@ -1415,6 +1414,7 @@ static int usb_ss_gadget_ep0_queue(struct usb_ep *ep,
 			gadget_writel(usb_ss, &usb_ss->regs->ep_cmd,
 			EP_CMD__ERDY__MASK | EP_CMD__REQ_CMPL__MASK);
 
+		cdns_prepare_setup_packet(usb_ss);
 		request->actual = 0;
 		usb_ss->status_completion_no_call = true;
 		usb_ss->pending_status_request = request;
-- 
1.7.9.5

